### DISCUSSION TITLE
LeftSection for inputs or how to put label inside InputWrapper

### DISCUSSION THREAD

**Discussion by 7iomka** [mantinedev/mantine#3253] [Q&A]
<img width="318" alt="image" src="https://user-images.githubusercontent.com/18473137/209826431-1a4a7ec1-0678-4f5f-ba74-978aac1efd99.png">

<img width="261" alt="image" src="https://user-images.githubusercontent.com/18473137/209826533-da0e37f3-87f5-4c4c-a663-c989af754576.png">

I want to make this type of input
It is hard to make a component wrapper around all types of inputs (number, text, etc) and to keep dynamic width for inlined label, and for this case if the label will be in the inputWrapper I can use display flex on it, and can have dynamic width.
Or, maybe we need to introduce a new prop called leftSection with static leftSectionWidth, so I can use it for my case.

**Comment by rtivital** [comment]
You can use `icon` with `iconWidth`

**Reply by 7iomka** [reply]
this way we have not accessible label for input

**Reply by rtivital** [reply]
You can put `label` element there

**Reply by 7iomka** [reply]
> You can put `label` element there

yes, but I need to pass all accesibility props (id, for, aria-*) by myself...

**Reply by rtivital** [reply]
Then pass it, what is the issue?

`<TextInput icon={<label {...yourPropsHere} />} />`

**Reply by 7iomka** [reply]
> Then pass it, what is the issue?
> 
> `<TextInput icon={<label {...yourPropsHere} />} />`

library generate automatically ids, so I need to make this for every input by myself...

**Reply by rtivital** [reply]
You can use `useId` hook exported from `@mantine/hooks` to connect label to the input.
